### DISCUSSION TITLE
How return custom data on browse view layout

### DISCUSSION THREAD

**Discussion by alancarlosml** [thedevdojo/voyager#5335] [Q&A]
## Version information

 - Laravel: v8.12
 - Voyager: v1.4
 - PHP: 8.0
 - Database: [MariaDB] [10.4]

## Description

I have two BREADS: Gym BREAD and Teacher BREAD.

In Gym BREAD browse view I have created a custom action button. The function of this button is to list all teachers that belongs to Gym model. To do so, I have another table gym_teacher that stores those ids - a many to many relationship.

I would like to know how I extend index function in TeacherController to return only that teachers.

I have tried some things suggested in Slack but did not work.

## Steps To Reproduce
1. Go to Gym browse view;
2. Click on 'Listar professores' button;
3. The data is returned witout Voyage browse layout.

## Expected behavior
After clicking on the "Listar professores" button, all teachers belonging to that gym should be returned in Voyager browse layout view.

## Screenshots
![image3](https://user-images.githubusercontent.com/413231/119194082-f2e64e00-ba58-11eb-9b5b-62bd45dd833a.png)
![image2](https://user-images.githubusercontent.com/413231/119194328-49538c80-ba59-11eb-91c4-4cdd6efd32a6.png)
![image](https://user-images.githubusercontent.com/413231/119194196-1a3d1b00-ba59-11eb-8dcd-e1463fc405b7.png)
![image](https://user-images.githubusercontent.com/413231/119194260-350f8f80-ba59-11eb-96ae-9e044654f532.png)

**Comment by Cyrille37** [comment]
Hi,
I did not found yet an easy way, just an open issue about similar subject : [[Feature] filtration and exporting feature #4216](https://github.com/the-control-group/voyager/issues/4216)

**Reply by Cyrille37** [reply]
In `VoyagerBaseController::index()` you can see some search criteria. If you construct a url (*a route should exists too*) like `http://127.0.0.1:8000/admin/posts?key=category_id&filter=equals&s=2` for the action it should do want you want. (*I'm using the `--with-dummy` Voyager installation*)

**Reply by Cyrille37** [reply]
With Voyager dummy data I add a `class PostUsersAction extends AbstractAction` like :
```
    // all stuff from https://voyager-docs.devdojo.com/customization/action-buttons
    // and those 2 methods more :

    public function getDefaultRoute()
    {
        return route('voyager.posts.index').'?key=author_id&filter=equals&s='.$this->data->id;
    }

    public function shouldActionDisplayOnDataType()
    {
        return $this->dataType->slug == 'users';
    }
```
Then on 'Users' browsing there is a "View posts" action button that point to "Posts browse" filtered by "author_id" with the row "user.id".

I'm just discovering Voyager, so perhaps another better way exists :-)

**Reply by emptynick** [reply]
I think the URL way is the easiest/most straight-forward
